### PR TITLE
WIP: Improve module import ergonomics

### PR DESCRIPTION
**Pull request by perimosocordiae**
I believe the plan is to support multiple types of importers, so I've started by making the generic Importer interface more agnostic to the type of module locator it accepts. For FileImporter, this should be a (relative) path to an Icarus file.

My main goal here is to make relative imports work.

### CODE REVIEW

**Comment by asoffer** [comment]
Issue #18 details some plans here. With those in mind, one way to proceed is to have the importer know how to turn a relative path into a full canonical path (based on compiler flags).

Feel free to gut/replace any of the file-name stuff. We should prefer strong-typing for these things most of the time, but I'm not certain the tools we have for those are the ones we want.

File: module/importer.cc [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+}
+
+}  // namespace
+
+frontend::CanonicalFileName ResolveModulePath(
+    frontend::CanonicalFileName const& module_path) {
+  // Respect absolute paths.
+  if (absl::StartsWith(module_path.name(), "/")) { return module_path; }
+  // Walk the entries in $ICARUS_MODULE_PATH to find a valid source file.
+  if (char* const lookup_paths = std::getenv("ICARUS_MODULE_PATH")) {
```
Comment (asoffer) [inline module/importer.cc:29]: Let's do this with compiler flags? Environment variable paths are a common solution to this but not ideal.

File: module/importer.h [hunk 2]
```diff
@@ -26,8 +30,10 @@ struct FileImporter : Importer {
     auto [id, mod, inserted] = ir::ModuleId::FromFile<ModuleType>(file_name);
     if (not inserted) { return id; }
 
+    frontend::CanonicalFileName const& module_path =
+        id.template filename<ModuleType>();
     if (auto maybe_file_src =
-            frontend::FileSource::Make(id.template filename<ModuleType>())) {
+            frontend::FileSource::Make(ResolveModulePath(module_path))) {
```
Comment (asoffer) [inline module/importer.h:36]: It seems like these are weirdly coupled, as opening the file source requires the file to exist in the same way that ResolveModulePath does.

This is fine for now, but I'm wondering this logic all belongs inside `FileSource::Make` (and if the whole source setup needs a better API).

**Review by asoffer — COMMENTED** [review]
Feel free/encouraged to commit whenever you feel comfortable. I won't officially "approve" anything unless you're specifically looking for it. Feel free to address my comments or not.

File: module/importer.h (hunk 2 above)
Comment (perimosocordiae) [reply module/importer.h:36]: I don't want to couple the generic FileSource logic too closely to the module path resolution logic. We could have a `FileSource::Make` overload that accepts an open `FILE*`, but that may be more complexity than it's worth.

File: module/importer.cc (hunk 1 above)
Comment (perimosocordiae) [reply module/importer.cc:29]: Compiler flags alone aren't very ergonomic, IMO, as they need to be specified each invocation. I think I can provide a compromise, though...

Comment (asoffer) [reply module/importer.cc]: My concern about environment variables is the statefulness. I think it's likely that both the compiler and interpreter are probably not invoked directly but rather by a build tool, reducing the benefit of environment variables.

**Pull request merged**